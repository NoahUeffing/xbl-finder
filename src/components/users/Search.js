import React, { Component } from "react";
import PropTypes from "prop-types";

export class Search extends Component {
  state = {
    text: "",
  };

  static propTypes = {
    searchFriends: PropTypes.func.isRequired,
    showAllFriends: PropTypes.func.isRequired,
    clearFriends: PropTypes.func.isRequired,
    showClear: PropTypes.bool.isRequired,
  };

  onSubmit = (e) => {
    e.preventDefault();
    this.props.searchFriends(this.state.text);
    //this.setState({ text: "" });
  };

  onChange = (e) => this.setState({ [e.target.name]: e.target.value });

  render() {
    const { showClear, clearFriends, showAllFriends } = this.props;
    return (
      <div>
        <form onSubmit={this.onSubmit} className="form">
          <input
            type="text"
            name="text"
            placeholder="Search Friends..."
            value={this.state.text}
            onChange={this.onChange}
          />
          <input
            type="submit"
            value="Search"
            className="btn btn-dark btn-block"
          />
        </form>
        <div>
          <div>
            <button
              className="btn btn-primary btn-block"
              onClick={showAllFriends}
            >
              Show All Friends
            </button>
          </div>
        </div>
        {showClear && (
          <button className="btn btn-dark btn-block" onClick={clearFriends}>
            Clear
          </button>
        )}
      </div>
    );
  }
}

export default Search;
