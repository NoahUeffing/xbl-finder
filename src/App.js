import React, { Fragment, Component } from "react";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import Navbar from "./components/layout/Navbar";
import Users from "./components/users/Users";
import Search from "./components/users/Search";
import GameSearch from "./components/games/GameSearch";
import Alert from "./components/layout/Alert";
import About from "./components/pages/About";
import MyInfo from "./components/pages/MyInfo";
import MyXbox360Games from "./components/pages/MyXbox360Games";
import MyXboxOneGames from "./components/pages/MyXboxOneGames";
import axios from "axios";
import "./App.css";

class App extends Component {
  state = {
    users: [],
    loading: false,
    alert: null,
    myInfo: [],
    newProfile: [],
    xboxOneGames: [],
    xbox360Games: [],
  };

  async componentDidMount() {
    this.setState({ loading: true });

    const friends = await axios.get(
      `https://xapi.us/v2/${process.env.REACT_APP_XAPI_USER_ID}/friends`,
      {
        headers: { "X-Auth": `${process.env.REACT_APP_XAPI_KEY}` },
      }
    );
    const selfInfo = await axios.get(
      `https://xapi.us/v2/${process.env.REACT_APP_XAPI_USER_ID}/gamercard`,
      {
        headers: { "X-Auth": `${process.env.REACT_APP_XAPI_KEY}` },
      }
    );
    const selfProfile = await axios.get(
      `https://xapi.us/v2/${process.env.REACT_APP_XAPI_USER_ID}/new-profile`,
      {
        headers: { "X-Auth": `${process.env.REACT_APP_XAPI_KEY}` },
      }
    );
    const xboxOne = await axios.get(
      `https://xapi.us/v2/${process.env.REACT_APP_XAPI_USER_ID}/xboxonegames`,
      {
        headers: { "X-Auth": `${process.env.REACT_APP_XAPI_KEY}` },
      }
    );
    const xbox360 = await axios.get(
      `https://xapi.us/v2/${process.env.REACT_APP_XAPI_USER_ID}/xbox360games`,
      {
        headers: { "X-Auth": `${process.env.REACT_APP_XAPI_KEY}` },
      }
    );

    this.setState({
      users: friends.data,
      myInfo: selfInfo.data,
      loading: false,
      newProfile: selfProfile.data,
      xboxOneGames: xboxOne.data.titles,
      xbox360Games: xbox360.data.titles,
    });
  }

  getMyInfo = async () => {
    this.setState({ loading: true });

    const res = await axios.get(
      `https://xapi.us/v2/${process.env.REACT_APP_XAPI_USER_ID}/gamercard`,
      {
        headers: { "X-Auth": `${process.env.REACT_APP_XAPI_KEY}` },
      }
    );
    this.setState({ myInfo: res.data, loading: false });
  };

  getNewProfile = async () => {
    this.setState({ loading: true });

    const res = await axios.get(
      `https://xapi.us/v2/${process.env.REACT_APP_XAPI_USER_ID}/new-profile`,
      {
        headers: { "X-Auth": `${process.env.REACT_APP_XAPI_KEY}` },
      }
    );
    this.setState({ newProfile: res.data, loading: false });
  };

  // Search Xbox Live Friends List

  searchFriends = async (text) => {
    this.setState({ loading: true });
    const res = await axios.get(
      `https://xapi.us/v2/${process.env.REACT_APP_XAPI_USER_ID}/friends`,
      {
        headers: { "X-Auth": `${process.env.REACT_APP_XAPI_KEY}` },
      }
    );
    var searchResult = res.data.filter((item) =>
      item.Gamertag.toUpperCase().includes(text.toUpperCase())
    );
    this.setState({ users: searchResult, loading: false });
  };

  // Show all friends
  showAllFriends = async () => {
    this.setState({ loading: true });

    const res = await axios.get(
      `https://xapi.us/v2/${process.env.REACT_APP_XAPI_USER_ID}/friends`,
      {
        headers: { "X-Auth": `${process.env.REACT_APP_XAPI_KEY}` },
      }
    );
    this.setState({ users: res.data, loading: false });
  };

  // Clear friends results

  clearFriends = () => this.setState({ users: [], loading: false });

  searchGames = async (text) => {
    this.setState({ loading: true });
    if (window.location.href.includes("myXboxOneGames")) {
      const res = await axios.get(
        `https://xapi.us/v2/${process.env.REACT_APP_XAPI_USER_ID}/xboxonegames`,
        {
          headers: { "X-Auth": `${process.env.REACT_APP_XAPI_KEY}` },
        }
      );
      var oneResult = res.data.titles.filter((item) =>
        item.name.toUpperCase().includes(text.toUpperCase())
      );
      this.setState({ xboxOneGames: oneResult, loading: false });
    }
    if (window.location.href.includes("myXbox360Games")) {
      const res = await axios.get(
        `https://xapi.us/v2/${process.env.REACT_APP_XAPI_USER_ID}/xbox360games`,
        {
          headers: { "X-Auth": `${process.env.REACT_APP_XAPI_KEY}` },
        }
      );
      var xbox360Result = res.data.titles.filter((item) =>
        item.name.toUpperCase().includes(text.toUpperCase())
      );
      this.setState({ xbox360Games: xbox360Result, loading: false });
    }
  };

  // Show all friends
  showAllGames = async () => {
    this.setState({ loading: true });

    if (window.location.href.includes("myXboxOneGames")) {
      const res = await axios.get(
        `https://xapi.us/v2/${process.env.REACT_APP_XAPI_USER_ID}/xboxonegames`,
        {
          headers: { "X-Auth": `${process.env.REACT_APP_XAPI_KEY}` },
        }
      );
      this.setState({ xboxOneGames: res.data.titles, loading: false });
    }
    if (window.location.href.includes("myXbox360Games")) {
      const res = await axios.get(
        `https://xapi.us/v2/${process.env.REACT_APP_XAPI_USER_ID}/xbox360games`,
        {
          headers: { "X-Auth": `${process.env.REACT_APP_XAPI_KEY}` },
        }
      );
      this.setState({ xbox360Games: res.data.titles, loading: false });
    }
  };

  // Clear friends results

  clearGames = () => {
    if (window.location.href.includes("myXboxOneGames")) {
      this.setState({ xboxOneGames: [], loading: false });
    }
    if (window.location.href.includes("myXbox360Games")) {
      this.setState({ xbox360Games: [], loading: false });
    }
  };

  // set alert
  setAlert = (msg, type) => {
    this.setState({ alert: { msg, type } });

    // TODO: change this to have an x button instead of timeout
    setTimeout(() => this.setState({ alert: null }), 5000);
  };

  render() {
    const {
      users,
      loading,
      myInfo,
      newProfile,
      xboxOneGames,
      xbox360Games,
    } = this.state;
    return (
      <Router>
        <div className="App">
          <Navbar />
          <div className="container">
            <Alert alert={this.state.alert} />
            <Switch>
              <Route
                exact
                path="/"
                render={(props) => (
                  <Fragment>
                    <Search
                      searchFriends={this.searchFriends}
                      showAllFriends={this.showAllFriends}
                      clearFriends={this.clearFriends}
                      showClear={users.length > 0 ? true : false}
                      setAlert={this.setAlert}
                    />
                    <Users loading={loading} users={users} />
                  </Fragment>
                )}
              />
              <Route exact path="/about" component={About} />
              <Route
                exact
                path="/myInfo"
                render={(props) => (
                  <MyInfo
                    loading={loading}
                    myInfo={myInfo}
                    newProfile={newProfile}
                  />
                )}
              />
              <Route
                exact
                path="/myXboxOneGames"
                render={(props) => (
                  <Fragment>
                    <GameSearch
                      searchGames={this.searchGames}
                      showAllGames={this.showAllGames}
                      clearGames={this.clearGames}
                      showClear={xboxOneGames.length > 0 ? true : false}
                      setAlert={this.setAlert}
                    />
                    <MyXboxOneGames
                      loading={loading}
                      games={xboxOneGames}
                      player={newProfile}
                    />
                  </Fragment>
                )}
              />
              <Route
                exact
                path="/myXbox360Games"
                render={(props) => (
                  <Fragment>
                    <GameSearch
                      searchGames={this.searchGames}
                      showAllGames={this.showAllGames}
                      clearGames={this.clearGames}
                      showClear={xbox360Games.length > 0 ? true : false}
                      setAlert={this.setAlert}
                    />
                    <MyXbox360Games
                      loading={loading}
                      games={xbox360Games}
                      player={myInfo}
                    />
                  </Fragment>
                )}
              />
            </Switch>
          </div>
        </div>
      </Router>
    );
  }
}

export default App;

export { App };
