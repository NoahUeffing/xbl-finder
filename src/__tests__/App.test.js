import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { App } from "../App";
import Users from "../components/users/Users";
import Search from "../components/users/Search";

describe("App", () => {
  it("renders without crashing given the required props", () => {
    const props = {
      loading: false,
      dispatch: jest.fn(),
      alert: null,
      users: [],
    };
    const wrapper = shallow(<App {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

it("renders the Search when the loading prop is false", () => {
  const props = {
    loading: false,
    dispatch: jest.fn(),
    alert: null,
    users: [],
  };
  const wrapper = shallow(<App {...props} />);
  expect(wrapper.find("Search").length).toBe(1);
});

it("renders the Navbar when the loading prop is false", () => {
  const props = {
    loading: false,
    dispatch: jest.fn(),
    alert: null,
    users: [],
  };
  const wrapper = shallow(<App {...props} />);
  expect(wrapper.find("Navbar").length).toBe(1);
});

it("renders the Users when the loading prop is false", () => {
  const props = {
    loading: false,
    dispatch: jest.fn(),
    alert: null,
    users: [],
  };
  const wrapper = shallow(<App {...props} />);
  expect(wrapper.find("Users").length).toBe(1);
});

it("sets the users prop as the `value` prop on the Users component", () => {
  const props = {
    loading: false,
    dispatch: jest.fn(),
    alert: null,
    users: [],
  };
  const wrapper = shallow(<App {...props} />);
  // Query for the Users component in the rendered output
  const UsersComponent = wrapper.find(Users);
  expect(UsersComponent.props().users).toEqual(props.users);
});

it("sets the Apps alert as the `value` prop on the App component", () => {
  const props = {
    loading: false,
    dispatch: jest.fn(),
    alert: null,
    users: [],
  };
  const wrapper = shallow(<App {...props} />);
  expect(wrapper.state("alert")).toEqual(props.alert);
});
