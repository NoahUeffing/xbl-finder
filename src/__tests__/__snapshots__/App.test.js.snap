// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing given the required props 1`] = `
<BrowserRouter>
  <div
    className="App"
  >
    <Navbar
      icon="fab fa-xbox"
      title="Aardvark Party's Xbox Companion"
    />
    <div
      className="container"
    >
      <Alert
        alert={null}
      />
      <Switch>
        <Route
          exact={true}
          path="/"
          render={[Function]}
        />
        <Route
          component={[Function]}
          exact={true}
          path="/about"
        />
        <Route
          exact={true}
          path="/myInfo"
          render={[Function]}
        />
        <Route
          exact={true}
          path="/myXboxOneGames"
          render={[Function]}
        />
        <Route
          exact={true}
          path="/myXbox360Games"
          render={[Function]}
        />
      </Switch>
    </div>
  </div>
</BrowserRouter>
`;
